### PR TITLE
Handle syntax checking and caching for nodes with dynamic values

### PR DESCRIPTION
**Pull request by jg18**
Nodes whose values can change are any of special argument, variable, or container data.

We should avoid caching values from such nodes and also avoid attempting "deep inspection" while checking SEXP syntax for things like ship/subsystem names.

We can, however, still do some type checking.

Should fix Issue #4226.

### CODE REVIEW

**Comment by jg18** [comment]
Since this PR also fixes the issue of (incorrectly) caching nodes that hold container data in some situations, it's a fix for containers and consequently should go into 22.2.

File: code/parse/sexp.cpp [hunk 1]
```diff
@@ -2464,8 +2465,8 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 				}
 				Assert(ship_node >= 0);
 
-				// we can't check special-arg ships
-				if (Sexp_nodes[ship_node].flags & SNF_SPECIAL_ARG_IN_NODE)
+				// we can't check nodes whose value can change
+				if (can_node_value_change(node))
```
Comment (jg18) [inline code/parse/sexp.cpp:2469]: I know I should check both here and in similar cases that the node is specifically one of a special argument, string variable, or string container data before skipping the check. I'll add that before I mark the PR as ready for review.

**Comment by jg18** [comment]
Passes manual test.

I'm able to open in FRED the mission @MjnMixael posted on Issue #4226 without error.

**Review by Goober5000 — APPROVED** [review]
This looks good. 👍  One optional comment.

**Pull request merged**